packages/connect-agreement/src/models/Agreement.ts

```diff
@@ -147,7 +153,77 @@ export default class Agreement {
     callback?: SubscriptionCallback<StakingMovement[]>
   ): SubscriptionResult<StakingMovement[]> {
     return subscription<StakingMovement[]>(callback, (callback) =>
-      this.#connector.onStakingMovements(this.stakingId(tokenAddress, userAddress), this.#address, first, skip, callback)
+      this.#connector.onStakingMovements(this.stakingId(tokenAddress, userAddress), this.address, first, skip, callback)
+    )
+  }
+
+  actionId(actionNumber: string): string {
+    return `${this.address.toLowerCase()}-action-${actionNumber}`
+  }
+
+  async action(actionNumber: string): Promise<Action> {
+    const action = await this.tryAction(actionNumber)
+    if (!action) {
+      throw Error(`Could not find given action number ${actionNumber}`)
+    }
+    return action as Action
+  }
+
+  async tryAction(actionNumber: string): Promise<Action | null> {
+    return this.#connector.action(this.actionId(actionNumber))
+  }
+
+  onAction(acitonNumber: string, callback?: SubscriptionCallback<Action | null>): SubscriptionResult<Action | null> {
+    return subscription<Action | null>(callback, (callback) =>
+      this.#connector.onAction(this.actionId(acitonNumber), callback)
     )
   }
+
+  sign(signerAddress: string, versionNumber: string): Promise<ForwardingPath> {
```

Comment:
I wonder if the `signerAddress` should be optional and part of a similar `PathOptions` object like we define here: https://github.com/aragon/connect/issues/132.

We can try to get an account from the provider or otherwise error if none was provided as an option?